### PR TITLE
Feature/xpt2046

### PR DESCRIPTION
**Pull request by ctacke**

### CODE REVIEW

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
-    public int X { get; set; }
-    public int Y { get; set; }
-}
-
 /// <summary>
 /// Represents a circle in the user interface.
 /// </summary>
 public class Circle : ThemedControl
 {
     private Color _foreColor;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs:8]: you heard the boss - underscores need to go

File: Source/Meadow.Foundation.Peripherals/ICs.IOExpanders.Ftxxxx/Samples/Ft232h_Sample/Ft232h_Sample.csproj [hunk 2]
```diff
@@ -7,12 +7,6 @@
     <Nullable>enable</Nullable>
   </PropertyGroup>
 
-  <ItemGroup>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/ICs.IOExpanders.Ftxxxx/Samples/Ft232h_Sample/Ft232h_Sample.csproj:10]: I assume this is intentional?

File: Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <TargetFramework>netstandard2.1</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AssemblyName>Xpt2046</AssemblyName>
+    <Company>Wilderness Labs, Inc</Company>
+    <PackageProjectUrl>http://developer.wildernesslabs.co/Meadow/Meadow.Foundation/</PackageProjectUrl>
+    <PackageId>Meadow.Foundation.Sensors.Hid.Tsc2004</PackageId>
+    <RepositoryUrl>https://github.com/WildernessLabs/Meadow.Foundation</RepositoryUrl>
+    <PackageTags>Meadow.Foundation,touch,screen,display,capacitive,XPT2046</PackageTags>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <Description>Xpt2046 SPI 4-wire resistive touch screen controller</Description>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj:19]: remove "4-wire"

**Review by adrianstevens — CHANGES_REQUESTED** [review]
The sample needs to be cleaned up as well - folder names and such

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Samples/Ili9488Charts_Sample/Ili9488Charts_Sample.csproj [hunk 4]
```diff
@@ -15,10 +15,4 @@
     <ProjectReference Include="..\..\Driver\Graphics.MicroLayout.csproj" />
   </ItemGroup>
 
-  <ItemGroup>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Samples/Ili9488Charts_Sample/Ili9488Charts_Sample.csproj:18]: And here - intentional delete?

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs [hunk 5]
```diff
@@ -1,17 +1,13 @@
...
-public struct Coordinate2D
-{
-    public int X { get; set; }
-    public int Y { get; set; }
-}
-
 /// <summary>
 /// Represents a circle in the user interface.
 /// </summary>
 public class Circle : ThemedControl
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs:6]: I'm wondering now if circle makes sense as a control ... we should talk about how we should handle drawing primitives

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs [hunk 6]
```diff
@@ -84,10 +82,10 @@ protected override void OnDraw(MicroGraphics graphics)
     {
         if (ForeColor != Color.Transparent)
         {
-            var radius = (Right - Left) / 2;
-            var centerX = Left + radius;
-            var centerY = Top + radius;
-            graphics.DrawCircle(centerX, centerY, radius, ForeColor, IsFilled);
+            //var radius = (Right - Left) / 2;
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs:85]: dead code

File: Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Samples/Tsc2004_Sample/MeadowApp.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
+﻿using Meadow;
+using Meadow.Devices;
+using Meadow.Foundation.Graphics;
+using Meadow.Foundation.Sensors.Hid;
+using Meadow.Hardware;
+using Meadow.Peripherals.Displays;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Bbq10Keyboard_Sample
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Samples/Tsc2004_Sample/MeadowApp.cs:10]: I'm assuming this namespace is wrong in the Tsc2004 sample too

File: Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <PackageIcon>icon.png</PackageIcon>
+    <Authors>Wilderness Labs, Inc</Authors>
+    <TargetFramework>netstandard2.1</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AssemblyName>Xpt2046</AssemblyName>
+    <Company>Wilderness Labs, Inc</Company>
+    <PackageProjectUrl>http://developer.wildernesslabs.co/Meadow/Meadow.Foundation/</PackageProjectUrl>
+    <PackageId>Meadow.Foundation.Sensors.Hid.Tsc2004</PackageId>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj:15]: Update ID

File: Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    <RepositoryUrl>https://github.com/WildernessLabs/Meadow.Foundation</RepositoryUrl>
+    <PackageTags>Meadow.Foundation,touch,screen,display,capacitive,XPT2046</PackageTags>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <Description>Xpt2046 SPI 4-wire resistive touch screen controller</Description>
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include="..\..\..\icon.png" Pack="true" PackagePath="" />
+    <ProjectReference Include="..\..\..\..\..\Meadow.Contracts\Source\Meadow.Contracts\Meadow.Contracts.csproj" />
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj:23]: Remove logging and contracts references

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs (hunk 1 above)
Comment (ctacke) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs:8]: Doh!  Muscle memory

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs (hunk 5 above)
Comment (ctacke) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs:6]: I would have preferred an Elipse, but we don't have the ability to draw that.  It matches with the Box control.  But yes, we can definitely discuss.

File: Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Samples/Tsc2004_Sample/MeadowApp.cs (hunk 7 above)
Comment (ctacke) [reply Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Samples/Tsc2004_Sample/MeadowApp.cs:10]: LOL, I copied the sample project over, but never added it to the solution of actually updated it.  Doing that now.

File: Source/Meadow.Foundation.Peripherals/ICs.IOExpanders.Ftxxxx/Samples/Ft232h_Sample/Ft232h_Sample.csproj (hunk 2 above)
Comment (ctacke) [reply Source/Meadow.Foundation.Peripherals/ICs.IOExpanders.Ftxxxx/Samples/Ft232h_Sample/Ft232h_Sample.csproj:10]: Yes.  libmpsse.dll is no longer required for FT232, so it's just removing library cruft

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Samples/Ili9488Charts_Sample/Ili9488Charts_Sample.csproj (hunk 4 above)
Comment (ctacke) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Samples/Ili9488Charts_Sample/Ili9488Charts_Sample.csproj:18]: Yes.  libmpsse.dll is no longer required for FT232, so it's just removing library cruft

File: Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    <TargetFramework>netstandard2.1</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AssemblyName>Xpt2046</AssemblyName>
+    <Company>Wilderness Labs, Inc</Company>
+    <PackageProjectUrl>http://developer.wildernesslabs.co/Meadow/Meadow.Foundation/</PackageProjectUrl>
+    <PackageId>Meadow.Foundation.Sensors.Hid.Xpt2046</PackageId>
+    <RepositoryUrl>https://github.com/WildernessLabs/Meadow.Foundation</RepositoryUrl>
+    <PackageTags>Meadow.Foundation,touch,screen,display,capacitive,XPT2046</PackageTags>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <Description>Xpt2046 SPI, 4-wire resistive touch screen controller</Description>
```
Comment (adrianstevens) [inline Source/Meadow.Foundation.Peripherals/Sensors.Hid.Xpt2046/Driver/Sensors.Hid.Xpt2046.csproj:19]: remove ", 4-wire" for consistancy .... no other drivers state this and it's used in the peripherals table as the description

File: Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs (hunk 5 above)
Comment (adrianstevens) [reply Source/Meadow.Foundation.Libraries_and_Frameworks/Graphics.MicroLayout/Driver/Controls/Circle.cs:6]: nothing to block on of course

**Review by adrianstevens — APPROVED** [review]

**Pull request merged**